pkg/solana/fees/block_history.go

```diff
@@ -133,5 +144,105 @@ func (bhe *blockHistoryEstimator) calculatePrice(ctx context.Context) error {
 		"slot", block.ParentSlot+1,
 		"count", len(feeData.Prices),
 	)
+
+	return nil
+}
+
+func (bhe *blockHistoryEstimator) calculatePriceFromMultipleBlocks(ctx context.Context, desiredBlockCount uint64) error {
+	// fetch client
+	c, err := bhe.client.Get()
+	if err != nil {
+		return fmt.Errorf("failed to get client: %w", err)
+	}
+
+	// Fetch the latest slot
+	currentSlot, err := c.SlotHeight(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get current slot: %w", err)
+	}
+
+	// Determine the starting slot for fetching blocks
+	if currentSlot < desiredBlockCount {
+		return fmt.Errorf("current slot is less than desired block count")
+	}
+	startSlot := currentSlot - desiredBlockCount + 1
+
+	// Fetch the last confirmed block slots
+	confirmedSlots, err := c.GetBlocksWithLimit(ctx, startSlot, desiredBlockCount)
+	if err != nil {
+		return fmt.Errorf("failed to get blocks with limit: %w", err)
+	}
+
+	// limit concurrency (avoid hitting rate limits)
+	semaphore := make(chan struct{}, 10)
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	allPrices := make([]ComputeUnitPrice, 0, desiredBlockCount)
+
+	// Iterate over the confirmed slots in reverse order to fetch most recent blocks first
+	// Iterate until we run out of slots
+	for i := len(*confirmedSlots) - 1; i >= 0; i-- {
+		slot := (*confirmedSlots)[i]
+
+		wg.Add(1)
+		go func(s uint64) {
+			defer wg.Done()
+			semaphore <- struct{}{}
+			defer func() { <-semaphore }()
+
+			// Fetch the block details
+			block, errGetBlock := c.GetBlock(ctx, s)
+			if errGetBlock != nil || block == nil {
+				// Failed to get block at slot || no block found at slot: skip.
+				return
+			}
+
+			// Parse the block to extract compute unit prices
+			feeData, errParseBlock := ParseBlock(block)
+			if errParseBlock != nil {
+				// Failed to parse block at slot: skip.
```

Comment:
Same here. Might be worth adding an error log in case there's some compatibility issue that comes up so we know what's breaking.